llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -10225,8 +10225,11 @@ SDValue TargetLowering::LowerToTLSEmulatedModel(const GlobalAddressSDNode *GA,
 
   ArgListTy Args;
   ArgListEntry Entry;
-  std::string NameString = ("__emutls_v." + GA->getGlobal()->getName()).str();
-  Module *VariableModule = const_cast<Module*>(GA->getGlobal()->getParent());
+  const GlobalValue *GV = GA->getGlobal();
+  GV = GV->stripPointerCastsAndAliases()
+	   ? cast<GlobalValue>(GV->stripPointerCastsAndAliases()) : GV;
```

Comment:
GV = GA->getGlobal()->stripPointerCastsAndAliases(), nothing is conditional

Reply:
ok, fixed